### PR TITLE
Copy shared files from release

### LINKED ISSUE #2069: Confusion about shared_dirs and Laravel storage folder
First, I really enjoy and appreciate Deployer. Thank you so much for making it.  It has been working well for me.

The only problem I've had recently is that now in my Laravel 7 project I commit to git some special files into my `storage/files` directory, and then those don't seem available after deployment.

I think the problem happens because of https://github.com/deployphp/deployer/blob/29a643b62389542394c7b254f6c913f7191132ba/recipe/deploy/shared.php

I'm not sure that this is a bug... but maybe just a lack of documentation.

I probably just don't know what this means: https://deployer.org/docs/configuration.html#shared_dirs

In my config, my `shared_dirs` is still empty.

I think a lot of other people have been confused about similar points too (and I still don't understand):

- https://github.com/deployphp/deployer/issues/572
- https://github.com/deployphp/deployer/issues/875
- https://github.com/deployphp/deployer/issues/957
- https://github.com/deployphp/deployer/issues/1593
- https://github.com/deployphp/deployer/issues/1647

Can you please clarify what `shared_files` and `shared_dirs` mean and whether they're related to my problem? Or maybe I'm accidentally not following Laravel patterns?

Thanks!

**Environment**
* Deployer version: I'm not sure how to tell
* PHP version: PHP 7.4.5 (cli)
* Deployment target(s) OS: Ubuntu

### PR DESCRIPTION
**Pull request by aryanjaya**
fix deployer cannot sync newer files or directories in shared directory

- [x] Bug fix #2069 #2377
- [ ] New feature?
- [ ] BC breaks?
- [ ] Deprecations?
- [ ] Tests added?
- [x] Changelog updated?

### CODE REVIEW

**Comment by aryanjaya** [comment]
Hi,

I want to try add some testing for this change and I already read some test files
but I still don't know where to start
any advice or reference where I can find an example?

Thanks

**Comment by antonmedv** [comment]
Sure, you can take a look at these test cases and do something similar: 
https://github.com/deployphp/deployer/tree/master/test/joy

File: recipe/deploy/shared.php [hunk 1]
```diff
@@ -27,10 +27,11 @@
...
             run("mkdir -p $sharedPath/$dir");
-            // If release contains shared dir, copy that dir from release to shared.
-            if (test("[ -d $(echo {{release_path}}/$dir) ]")) {
-                run("cp -r{$copyVerbosity} {{release_path}}/{$dir} {$sharedPath}/" . dirname(parse($dir)));
-            }
+        }
+
+        // If release contains shared dir, copy that dir from release to shared.
+        if (test("[ -d $(echo {{release_path}}/$dir) ]")) {
+            run("cp -ru{$copyVerbosity} {{release_path}}/{$dir} {$sharedPath}/" . dirname(parse($dir)));
```
Comment (antonmedv) [inline recipe/deploy/shared.php:34]: This is a really crucial part and needs to be tested.

**Review by antonmedv — CHANGES_REQUESTED** [review]

File: recipe/deploy/shared.php (hunk 1 above)
Comment (aryanjaya) [reply recipe/deploy/shared.php:34]: Sure, I still make the test.

Is there anything specific that I need to pay attention to in making the test?

Comment (antonmedv) [reply recipe/deploy/shared.php:34]: I think new test should cover use case of shared copy. Look at `joy` tests.

**Review by BryanLaura — APPROVED** [review]

**Comment by antonmedv** [comment]
@aryanjaya any update on this?

**Comment by aryanjaya** [comment]
@antonmedv Sorry for not updating this pull request yet. I still don't know how to make a test case for this issue.

**Comment by antonmedv** [comment]
What about https://github.com/deployphp/deployer/tree/master/test/joy joy testing?)

**Review by BryanLaura — APPROVED** [review]

**Comment by antonmedv** [comment]
Looking for someone who can take and improve this PR with tests? Any new contributors?

**Pull request closed without merging**